src/schedule/MongoSchedule.ts

```diff
@@ -27,7 +27,21 @@ export class MongoSchedule extends Schedule {
     protected readonly scheduleId: string,
     protected readonly connection: Connection,
     pingIntervalMs: number,
+    maxPingRetries: number,
+    retryIntervalMs: number,
   ) {
+    if (!Number.isFinite(pingIntervalMs) || pingIntervalMs < 1) {
+      throw new Error('Error: pingIntervalMs must be a positive number');
+    }
+
+    if (!Number.isFinite(maxPingRetries) || maxPingRetries < 1) {
+      throw new Error('Error: maxPingRetries must be a positive number');
+    }
+
+    if (!Number.isFinite(retryIntervalMs) || retryIntervalMs < 1) {
```

Comment:
There's also a max timeout delay for node. We should check for that as well. Take a look at `maxNodeTimeoutDelay` in this repo and how it's used.